### PR TITLE
SAIC-17: Check bandwidth

### PR DESCRIPTION
**Pull request by miarmak**
- Add 'check_bandwidth' action;
- Add 'initial_check' section to config;
- Add 'timer' function to cloudferrylib/utils/utils.py to measure python
  functions execution time.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/125/

File: cloudferrylib/os/actions/check_bandwidth.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                                     '/tmp/')
+
+        scp_download = cmd_cfg.scp_cmd(local_file_path,
+                                       'root',
+                                       self.cloud.cloud_config.cloud.ssh_host,
+                                       self.cloud.cloud_config.cloud.temp,
+                                       '')
+
+        # I guess it can be moved from here in other place
+        check_dir_cmd = "if [ ! -d {0} ]; then mkdir {0}; fi".format(
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_bandwidth.py]: `mkdir -p <path>` does this for you (see `mkdir --help`), so I guess you can move this command to `cmd_cfg.py`

**Comment by roman-verchikov** [comment]
Please rebase and fix inline

**Comment by miarmak** [comment]
- rebased
- change check_temp_dir to 'mkdir -p <path>'
- add random temp_file_name generation

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.   http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/138/

File: cloudferrylib/utils/cmd_cfg.py [hunk 2]
```diff
@@ -27,5 +28,8 @@
 ssh_cmd_port = base_ssh_cmd("-oStrictHostKeyChecking=no -p %s %s '%s'")
 dd_cmd_of = BC("dd bs=%s of=%s")
 dd_cmd_if = BC("dd bs=%s if=%s")
+dd_full = BC('dd if=%s of=%s bs=%s count=%s seek=%sM')
 gunzip_cmd = BC("gunzip")
-gzip_cmd = BC("gzip -%s -c %s")
\ No newline at end of file
+gzip_cmd = BC("gzip -%s -c %s")
+scp_cmd = BC('scp %s %s@%s:%s %s')
+rm_cmd = BC('rm -rf %s')
```
Comment (roman-verchikov) [inline cloudferrylib/utils/cmd_cfg.py]: You don't really need `-r` in `rm`, you're only removing one file. Let's modify it to `rm -f %s`.

**Comment by roman-verchikov** [comment]
Just one very minor inline (no more than 5 minutes of work), otherwise looks good, +1

File: cloudferrylib/utils/cmd_cfg.py (hunk 2 above)
Comment (miarmak) [inline cloudferrylib/utils/cmd_cfg.py]: Done

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.   http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/151/

**Comment by miarmak** [comment]
Done. Thanks for review

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
We need one more +1 to merge

**Comment by miarmak** [comment]
Rebased

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/172/

**Comment by roman-verchikov** [comment]
Looks like one more rebase is needed

**Comment by miarmak** [comment]
Rebased

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/179/

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
Please rebase

**Comment by miarmak** [comment]
Rebased

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/198/

**Pull request merged**